aws/resource_aws_rds_cluster.go

```diff
@@ -374,11 +411,12 @@ func resourceAwsRDSClusterCreate(d *schema.ResourceData, meta interface{}) error
 
 	if _, ok := d.GetOk("snapshot_identifier"); ok {
 		opts := rds.RestoreDBClusterFromSnapshotInput{
-			DBClusterIdentifier: aws.String(d.Get("cluster_identifier").(string)),
-			Engine:              aws.String(d.Get("engine").(string)),
-			EngineMode:          aws.String(d.Get("engine_mode").(string)),
-			SnapshotIdentifier:  aws.String(d.Get("snapshot_identifier").(string)),
-			Tags:                tags,
+			DBClusterIdentifier:  aws.String(d.Get("cluster_identifier").(string)),
+			Engine:               aws.String(d.Get("engine").(string)),
+			EngineMode:           aws.String(d.Get("engine_mode").(string)),
+			ScalingConfiguration: expandRdsScalingConfiguration(d.Get("scaling_configuration").([]interface{})),
```

Comment:
Should we do a d.GetOk since `scaling_configuration` is optional.

Reply:
In many cases, we allow the expand functions to `return nil` when it is not defined in the Terraform configuration -- this saves the extra logic of `d.GetOk()` each time we invoke these functions as well as prevents potential panics relating to zero elements or the first element being empty.